### PR TITLE
Filter input file-in-file-group

### PR DESCRIPTION
**Pull request by ascobie**
Fix: #1209

If you supply a comma separated list of wildcards in the file-in-file-group template parameter, you get to see this in the file picker.

```
        "blendFile": {
            "type": "string",
            "metadata": {
                "description": "The Blender scene file to be rendered",
                "advancedType": "file-in-file-group",
                "dependsOn": "inputData",
                "wildcards": "*.blend"
            }
        },
```

image

Checkbox allows for a fetch all operation against the container.

Also changed StringUtils.matchWildcard to match *word* and added tests for it.

### CODE REVIEW

File: app/components/data/shared/cloud-file-picker/cloud-file-picker.component.ts [hunk 1]
```diff
@@ -28,6 +28,12 @@ export class CloudFilePickerComponent implements ControlValueAccessor, OnChanges
      */
     @Input() public containerId: string;
 
+    /**
+     * Passed in if we want to client side filter the results to only show
+     * files that end with this suffix.
+     */
+    @Input() public suffixFilter: string;
```
Comment (timotheeguerin) [inline app/components/data/shared/cloud-file-picker/cloud-file-picker.component.ts]: I would maybe here instead have the same interface as openDialog from electron 

```js
filters: [
    { name: 'All Files', extensions: ['*'] },
    { name: 'Images', extensions: ['jpg', 'png', 'gif'] },
    { name: 'Movies', extensions: ['mkv', 'avi', 'mp4'] }
],
````

File: app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.component.ts [hunk 2]
```diff
@@ -56,8 +59,14 @@ export class CloudFilePickerDialogComponent {
...
     }
 
-    public updatePickedFile(file: string) {
-        this.pickedFile = file;
+    public ngOnInit() {
+        if (this.wildcards) {
+            this.pickedFilter = this.wildcards;
+            this.optionFilters = [
+                { label: "All Files", value: "" },
+                { label: `Wildcards (${this.wildcards})`, value: this.wildcards },
```
Comment (timotheeguerin) [inline app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.component.ts]: I would maybe not say Wildcards in front just the file extension

File: app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.html [hunk 3]
```diff
@@ -1,15 +1,32 @@
 <bl-simple-form [submit]="submit" (done)="close()" [containerRef]="dialogRef" [multiUse]="false" actionName="Pick">
     <div *ngIf="container">
-        <bl-blob-files-browser #blobList [storageAccountId]="storageAccountId" [container]="container.id" [fileExplorerConfig]="fileExplorerConfig" [activeFile]="pickedFile"
+        <bl-blob-files-browser #blobList
+            [storageAccountId]="storageAccountId"
+            [container]="container.id"
+            [fileExplorerConfig]="fileExplorerConfig"
+            [activeFile]="pickedFile"
+            [filenameWildcards]="pickedFilter"
```
Comment (timotheeguerin) [inline app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.html]: how about just filter

File: app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.html [hunk 4]
```diff
@@ -1,15 +1,32 @@
...
+            [recursiveFetch]="recursiveFetch"
```
Comment (timotheeguerin) [inline app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.html]: and here either recursive or loadAll

File: app/styles/vendor/material-theme.scss [hunk 5]
```diff
@@ -117,6 +117,9 @@ mat-dialog-container.mat-dialog-container {
     width: auto !important;
     height: auto !important;
     max-height: 80vh;
+    // have the children control the overflow to get DDL to show in dialog footer.
+    // there might be another way but havent found it yet.
+    overflow: visible !important;
```
Comment (timotheeguerin) [inline app/styles/vendor/material-theme.scss]: hhm, yeah that one you should wait for select v2. That's one of the problem I was talking about

Comment (ascobie) [reply app/styles/vendor/material-theme.scss]: This doesn't appear to affect anything else at this stage. As soon as select V2 goes in i am happy to remove this as its a bit of a dirty hack.

File: app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.html (hunk 4 above)
Comment (ascobie) [reply app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.html]: yeah, always looked at this and never really liked the term. will change.

File: app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.component.ts (hunk 2 above)
Comment (ascobie) [reply app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.component.ts]: What about "(*.blend,*.other) Files". Or just leave out the "Files" as well.

Comment (timotheeguerin) [reply app/components/data/shared/cloud-file-picker/cloud-file-picker-dialog.component.ts]: Yeah files would work too I feel

File: app/styles/vendor/material-theme.scss (hunk 5 above)
Comment (ascobie) [reply app/styles/vendor/material-theme.scss]: removed this hack now as select changes have been merged in.

**Pull request merged**